File: spacechat-backend/src/main/java/com/dimka/spacechat/entity/Call.java
package com.dimka.spacechat.entity;

import lombok.Data;
import lombok.experimental.Accessors;

import java.util.HashMap;
import java.util.Map;

@Accessors(chain = true)
@Data
public class Call {

    private String id;
    private String name;
    private Map<Long, UserSession> participants = new HashMap<>();
}

File: spacechat-frontend/src/entity/Call.ts
export class Call {

  id: string;
  name: string;
  participants: [string]
  status: string;
  screenShareOwner: string;
  selfScreenShare: boolean;

  constructor(id: string, status: string, name: string, participants: [string]) {
    this.id = id;
    this.status = status;
    this.name = name;
    this.participants = participants;
    this.selfScreenShare = false;
  }
}
